### PR TITLE
Add WebhooksController for processing deposits

### PR DESCRIPTION
**Pull request by ysv**

### CODE REVIEW

**Comment by ysv** [comment]
@yivo review please

File: app/controllers/webhooks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Service triggers webhook due to incoming bitcoin deposits.
+# Than we process in and send to daemon for customer's balance updating.
+class WebhooksController < ApplicationController
+
+  before_action :auth_anybody!
+
+  def tx
```
Comment (yivo) [inline app/controllers/webhooks_controller.rb]: Use single quotes. Write code which handles situation when invalid or no data is given (respond 422). Don't use obsolete hash syntax. Change URL & action to `/webhooks/btc/tx_created => post btc_tx_created`. Respond with 204.

File: config/routes.rb [hunk 2]
```diff
@@ -83,6 +83,8 @@ def draw(routes_name)
    post '/pusher/auth', to: 'pusher#auth'
   end
 
+  post '/webhooks/tx' => 'webhooks#tx', as: :deposit_webhook
```
Comment (yivo) [inline config/routes.rb]: Remove `:as`.

**Review by yivo — CHANGES_REQUESTED** [review]

File: app/controllers/webhooks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      AMQPQueue.enqueue(:deposit_coin, txid: webhook_params[:hash], channel_key: webhook_params[:ccy])
+      render json: { status: :no_content }
+    else
+      render json: { status: :unprocessable_entity }
+    end
+  end
+
+  private
+  def webhook_params
+    params.permit(:type, :hash, :ccy)
```
Comment (yivo) [inline app/controllers/webhooks_controller.rb]: @yssavchuk don't use permit unless you are dealing with models

File: app/controllers/webhooks_controller.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+# Than we process in and send to daemon for customer's balance updating.
+class WebhooksController < ApplicationController
+
+  before_action :auth_anybody!
+  skip_before_filter :verify_authenticity_token
+
+  def tx_created
+    if webhook_params[:type] == "transaction" && webhook_params[:hash].present? && Currency.coins.map(&:code)
+      AMQPQueue.enqueue(:deposit_coin, txid: webhook_params[:hash], channel_key: webhook_params[:ccy])
+      render json: { status: :no_content }
```
Comment (yivo) [inline app/controllers/webhooks_controller.rb]: @yssavchuk don't render unneeded JSON. Use head :no_content.

File: app/controllers/webhooks_controller.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Service triggers webhook due to incoming bitcoin deposits.
+# Than we process in and send to daemon for customer's balance updating.
+class WebhooksController < ApplicationController
+
+  before_action :auth_anybody!
+  skip_before_filter :verify_authenticity_token
+
+  def tx_created
+    if webhook_params[:type] == "transaction" && webhook_params[:hash].present? && Currency.coins.map(&:code)
```
Comment (yivo) [inline app/controllers/webhooks_controller.rb]: @yssavchuk use single quotes.

File: app/controllers/webhooks_controller.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+      AMQPQueue.enqueue(:deposit_coin, txid: webhook_params[:hash], channel_key: webhook_params[:ccy])
+      render json: { status: :no_content }
+    else
+      render json: { status: :unprocessable_entity }
+    end
+  end
+
+  private
+  def webhook_params
```
Comment (yivo) [inline app/controllers/webhooks_controller.rb]: @yssavchuk unneeded

File: app/models/currency.rb [hunk 7]
```diff
@@ -25,6 +25,10 @@ def self.assets(code)
     find_by_code(code)[:assets]
   end
 
+  def self.coins
+    @coins ||= Currency.where(coin: true)
```
Comment (yivo) [inline app/models/currency.rb:29]: @yssavchuk ok for now but we will remove it when dynamic configuration arrives.

File: app/controllers/webhooks_controller.rb [hunk 8]
```diff
@@ -0,0 +1,21 @@
+# Service triggers webhook due to incoming bitcoin deposits.
```
Comment (yivo) [inline app/controllers/webhooks_controller.rb]: @yssavchuk the descriptions must be updated.

File: app/controllers/webhooks_controller.rb [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+# Than we process in and send to daemon for customer's balance updating.
+class WebhooksController < ApplicationController
+
+  before_action :auth_anybody!
+  skip_before_filter :verify_authenticity_token
```
Comment (yivo) [inline app/controllers/webhooks_controller.rb]: @yssavchuk use skipe_before_action

File: app/controllers/webhooks_controller.rb (hunk 5 above)
Comment (yivo) [inline app/controllers/webhooks_controller.rb]: @yssavchuk what it does? `&& Currency.coins.map(&:code)`

**Review by yivo — CHANGES_REQUESTED** [review]

**Review by yivo — APPROVED** [review]

**Pull request merged**